src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -812,6 +812,8 @@ private Task WriteSuffix()
             var responseHeaders = FrameResponseHeaders;
             var hasConnection = responseHeaders.HasConnection;
             var connectionOptions = hasConnection ? FrameHeaders.ParseConnection(responseHeaders.HeaderConnection) : ConnectionOptions.None;
+            var hasTransferEncoding = responseHeaders.HasTransferEncoding;
+            var transferCoding = hasTransferEncoding ? FrameHeaders.GetFinalTransferCoding(responseHeaders.HeaderTransferEncoding) : TransferCoding.None;
```

Comment:
Just realized I don't even have to check `Count` - the `foreach` simply won't iterate over anything and those methods will return `None`.